client/src/utilities/startup.rs

```diff
@@ -56,9 +57,9 @@ impl SimpleState for MainState {
 pub fn start_game(
     resources: PathBuf,
     game_data_builder: GameDataBuilder<'static, 'static>,
-    state: Option<Box<dyn SimpleState>>,
+    state: Option<Box<dyn State<GameData<'static, 'static>, ExtendedStateEvent>>>,
 ) {
-    let mut game: Application<'_, GameData<'_, '_>> =
+    let mut game: CoreApplication<'_, GameData, ExtendedStateEvent, ExtendedStateEventReader> =
```

Comment:
Could the black screen be caused here?

Reply:
Not sure, but that one is required, and it seems that the resources have not been successfully loaded.